kernel/arch/dreamcast/hardware/g1ata.c

```diff
@@ -394,14 +401,14 @@ int g1_ata_read_chs(uint16_t c, uint8_t h, uint8_t s, size_t count,
     g1_ata_wait_bsydrq();
 
     while(count) {
-        nsects = count > 255 ? 255 : (uint8_t)count;
+        nsects = count > 256 ? 256 : (uint16_t)count;
         count -= nsects;
 
         g1_ata_select_device(G1_ATA_SLAVE | (h & 0x0F));
 
         /* Write out the number of sectors we want as well as the cylinder and
            sector. */
-        OUT8(G1_ATA_SECTOR_COUNT, nsects);
+        OUT8(G1_ATA_SECTOR_COUNT, nsects == 256 ? 0 : nsects);
```

Comment:
I think it was correct before, because of the cast to 8 bits.

You can make it explicit if you want:
`OUT8(G1_ATA_SECTOR_COUNT, (uint8_t)nsects);`